internal/controller/clusterorder_webhook.go

```diff
@@ -6,40 +6,102 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"sync"
 	"time"
 
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 
 	cloudkitv1alpha1 "github.com/innabox/cloudkit-operator/api/v1alpha1"
 )
 
-func triggerWebHook(ctx context.Context, url string, instance *cloudkitv1alpha1.ClusterOrder) error {
+type InflightRequest struct {
+	createTime time.Time
+}
+
+var (
+	inflightRequestsLock *sync.RWMutex              = &sync.RWMutex{}
+	inflightRequests     map[string]InflightRequest = make(map[string]InflightRequest)
+)
+
+func checkForExistingRequest(url string, minimumRequestInterval time.Duration) time.Duration {
+	var delta time.Duration
+
+	inflightRequestsLock.RLock()
+	request, ok := inflightRequests[url]
+	if ok {
+		delta = time.Since(request.createTime)
+		if delta >= minimumRequestInterval {
+			delta = 0
+		}
+	}
+	inflightRequestsLock.RUnlock()
+	purgeExpiredRequests(minimumRequestInterval)
+	return delta
+}
+
+func addInflightRequest(url string, minimumRequestInterval time.Duration) {
+	inflightRequestsLock.Lock()
+	inflightRequests[url] = InflightRequest{
+		createTime: time.Now(),
+	}
+	inflightRequestsLock.Unlock()
+	purgeExpiredRequests(minimumRequestInterval)
+}
+
+func purgeExpiredRequests(minimumRequestInterval time.Duration) {
+	var expiredRequests = make([]string, 3)
```

Comment:
Just initialize this empty. Otherwise you'll make it have a size of 3, and the first three items will each be an empty string (the "zero value" for type string).

```
expiredRequests := []string{}
```